binderhub/builder.py

```diff
@@ -169,6 +181,8 @@ def escape(s):
         if ref is None:
             await self.fail("Could not resolve ref for %s. Double check your URL." % key)
             return
+
+        # generate a complete build name `build-{user}-{repo}-{ref}`
```

Comment:
It's {user}-{repo}-{ref} only for github, it varies for other providers. There's a big docstring on _generate_build_api explaining what it tries to do...